### PR TITLE
🌱 make Cluster API dev tooling to work on arm64

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
make Cluster API dev tooling to work on arm64

### CODE REVIEW

File: scripts/ci-e2e-lib.sh [hunk 1]
```diff
@@ -18,21 +18,22 @@
 capi:buildDockerImages () {
   # Configure provider images generation;
   # please ensure the generated image name matches image names used in the E2E_CONF_FILE
+  ARCH="$(go env GOARCH)"
   export REGISTRY=gcr.io/k8s-staging-cluster-api
   export TAG=dev
-  export ARCH=amd64
+  export ARCH
```
Comment (vincepri) [inline scripts/ci-e2e-lib.sh:24]:
```suggestion
  export ARCH="$(go env GOARCH)"
```

Nit

Comment (fabriziopandini) [reply scripts/ci-e2e-lib.sh:24]: shell checker was complying when I originally inlined export and $(), this was the suggested solution

**Review by vincepri — APPROVED** [review]
/milestone v1.1

File: scripts/ci-e2e-lib.sh (hunk 1 above)
Comment (sbueringer) [reply scripts/ci-e2e-lib.sh:24]: The problem is that the error is masked if you have it in one line

**Pull request merged**